llvm/lib/Transforms/IPO/SampleProfile.cpp

```diff
@@ -1713,13 +1713,15 @@ void SampleProfileLoader::generateMDProfMetadata(Function &F) {
       // if needed. Sample counts in profiles are 64-bit unsigned values,
       // but internally branch weights are expressed as 32-bit values.
       if (Weight > std::numeric_limits<uint32_t>::max()) {
-        LLVM_DEBUG(dbgs() << " (saturated due to uint32_t overflow)");
+        LLVM_DEBUG(dbgs() << " (saturated due to uint32_t overflow)\n");
         Weight = std::numeric_limits<uint32_t>::max();
       }
       if (!SampleProfileUseProfi) {
         // Weight is added by one to avoid propagation errors introduced by
         // 0 weights.
-        Weights.push_back(static_cast<uint32_t>(Weight + 1));
+        Weights.push_back(static_cast<uint32_t>(
```

Comment:
I'm curious if this path is still beneficial for some use cases? Can we trust profi results and drop this hack altogether? @omern1

Reply:
you mean delete the non-profi inference path completely? some people might still want a fall back.

Reply:
i'm curious to learn about such use cases; are there some known examples where fallback is preferred?

Reply:
Hi @spupyrev, I can't speak for all users of the non-profi inference path but from our perspective that would require analysis to understand the impact of profi on our codebases.

I think this discussion can be had separately from this patch.

If you / @WenleiHe  / @MatzeB are happy with the current state of this patch can I please get an LGTM?

Reply:
Of course, I didn't plan to delay the patch, will stamp it. 
(Though if you care about perf and the experiments with profi aren't too time-consuming, please try it out and share the results with us. Ultimately it would be great to keep only one codepath here, but the change should be done carefully)